File: lab 6/Restaurants/Restaurants/Restaurant.swift
//
//  Restaurant.swift
//  Restaurants
//
//  Created by Jacob Brauchler on 4/8/17.
//  Copyright © 2017 JBrauchler. All rights reserved.
//

import Foundation
import Firebase

class Restaurant {
    var name: String
    var type: String
    var url: String
    
    init(newname: String, newurl: String, newtype: String){
        name = newname
        url = newurl
        type = newtype
    }
    
    init(snapshot: FIRDataSnapshot) {
        let snapshotValue = snapshot.value as! [String: String]
        name = snapshotValue["name"]!
        url = snapshotValue["url"]!
        type = snapshotValue["type"]!
    }
    
}

File: lab 9/Restaurants/app/src/main/java/com/myapps/jbrauchler/restaurants/Restaurant.java
package com.myapps.jbrauchler.restaurants;

import com.google.firebase.database.Exclude;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by jacobbrauchler on 4/27/17.
 */

public class Restaurant {
    private String name;
    private String type;
    private String url;

    public Restaurant(){
        // Default constructor required for calls to DataSnapshot.getValue(RecipeItem.class)
    }

    public Restaurant(String newName, String newType, String newURL){
        name = newName;
        type = newType;
        url = newURL;
    }

    public String getName(){
        return name;
    }

    public String getType() { return type; }

    public String geturl(){
        return url;
    }

    //used when writing to the database
    @Exclude
    public Map<String, Object> toMap(){
        HashMap<String, Object> result = new HashMap<>();
        result.put("name", name);
        result.put("type", type);
        result.put("url", url);
        return result;
    }

    //the string representation of a recipe name
    public String toString(){
        return this.name;
    }
}
